### ISSUE TITLE
Const constructor?

### ISSUE THREAD

**Issue by dunmatt** [dtolnay/semver#200]
Would you guys be open to a PR that makes `Version::new` const?

**Comment by iliana** [comment]
From https://github.com/steveklabnik/semver/pull/201#issuecomment-633354891:

> Hi there! I'd like to deal with `const` in a more holistic way across the codebase, so I won't be accepting this PR. Thank you though! This is an important use-case.

@steveklabnik What do you mean by a more holistic way? I'd like to help support this use case to remove a lazy_static dependency in a library I work on.

**Comment by steveklabnik** [comment]
I mean, looking at the entire API and deciding what should and should not be const.

but i am also working on re-writing the entire everythying, so maybe if it's a blocker for you, i should just accept that PR and cut a release to unblock things.

**Comment by iliana** [comment]
It's not a *blocker* by any means, it's a nice-to-have. :) But thank you!

**Comment by dtolnay** [comment]
Fixed in 1.0.0. https://docs.rs/semver/1.0.0-rc.1/semver/struct.Version.html#method.new
